### PR TITLE
TEN CLI

### PR DESCRIPTION
**Pull request by digitallysavvy**
1. Updated `Makefile` to reduce the commands to starting services
2. Updated Docker and DockerCompose to build for local
3. Created TEN cli and installer script (`install_cli.sh`) to simplify commands under a single API

### CODE REVIEW

File: ten [hunk 1]
```diff
@@ -0,0 +1,409 @@
+#!/Users/arlene/.ten-cli/venv/bin/python3
```
[the remark this answers is not in the document]
Comment (digitallysavvy) [reply ten:1]: i'll fix this shebang line

File: Dockerfile [hunk 2]
```diff
@@ -24,15 +25,25 @@ RUN apt-get clean && apt-get update && apt-get install -y --no-install-recommend
     python3-dev \
     jq \
     ca-certificates \
+    build-essential \
```
[the remark this answers is not in the document]
Comment (digitallysavvy) [reply Dockerfile:28]: ahh ok, I think there was some confusion around that. It wasn't explained anywhere in the docs, so if this is the case we should update docs to make clear. Once it's released we can make the appropriate updates.

**Comment by plutoless** [comment]
1. i'm using bash + iTerm2, it seems the install_cli.sh script will return me
"/usr/local/Cellar/jenv/0.5.4/libexec/libexec/../completions/jenv.bash:15: command not found: complete"
2. after running the script it will turn my shell to zsh..

**Comment by digitallysavvy** [comment]
> 1. i'm using bash + iTerm2, it seems the install_cli.sh script will return me
>    "/usr/local/Cellar/jenv/0.5.4/libexec/libexec/../completions/jenv.bash:15: command not found: complete"
> 2. after running the script it will turn my shell to zsh..

I will spend some time to update the logic for supporting the different shells

**Comment by plutoless** [comment]
> > 1. i'm using bash + iTerm2, it seems the install_cli.sh script will return me
> >    "/usr/local/Cellar/jenv/0.5.4/libexec/libexec/../completions/jenv.bash:15: command not found: complete"
> > 2. after running the script it will turn my shell to zsh..
> 
> I will spend some time to update the logic for supporting the different shells

i take a deeper look and it seems it's running my .bash_profile and it's giving above errors..

> > 1. i'm using bash + iTerm2, it seems the install_cli.sh script will return me
> >    "/usr/local/Cellar/jenv/0.5.4/libexec/libexec/../completions/jenv.bash:15: command not found: complete"
> > 2. after running the script it will turn my shell to zsh..
> 
> I will spend some time to update the logic for supporting the different shells

```
bash-3.2$ echo $(basename "$SHELL")
zsh
bash-3.2$
```

i think it's caused by incorrect shell detection. it's trying to use zsh run my bash_profile

**Comment by wangyoucao577** [comment]
> > 1. i'm using bash + iTerm2, it seems the install_cli.sh script will return me
> >    "/usr/local/Cellar/jenv/0.5.4/libexec/libexec/../completions/jenv.bash:15: command not found: complete"
> > 2. after running the script it will turn my shell to zsh..
> 
> I will spend some time to update the logic for supporting the different shells

Is it really necessary to install the tool? Maybe `./cli` run it is good enough, and can avoid issues if it updated.

**Comment by digitallysavvy** [comment]
Given the conversation in this thread, I will close this PR, make updates to the code and open a new PR when it's ready.

**Pull request closed without merging**